vyper/functions/functions.py

```diff
@@ -1224,6 +1227,61 @@ def _can_compare_with_uint256(operand):
     )
 
 
+@signature('decimal')
+def sqrt(expr, args, kwargs, context):
+    from vyper.functions.utils import (
+        generate_inline_function,
+    )
+    arg = args[0]
+    sqrt_code = """
+assert x >= 0.0
+z: decimal = (x + 1.0) / 2.0
+y: decimal = x
+
+for i in range(256):
+    if (z >= y):
+        break
+    y = z
+    z = (x / z + z) / 2.0
+    """
+
+    x_type = BaseType('decimal')
+    placeholder_copy = ['pass']
+    # Steal current position if variable is already allocated.
+    if arg.value == 'mload':
+        new_var_pos = arg.args[0]
+    # Other locations need to be copied.
+    else:
+        new_var_pos = context.new_placeholder(x_type)
```

Comment:
Maybe we can just use FREE_VAR_SPACE? Not sure if it will conflict with anything else

Reply:
FREE_VAR_SPACE will be overwritten, that's why we allocate new memory.